packages/ui/src/components/SelectInputV2/SelectBar.tsx

```diff
@@ -0,0 +1,337 @@
+  max-width: ${({ width }) => width}px;
+  overflow: scroll;
+`
+const StyledInputWrapper = styled(Stack, {
+  shouldForwardProp: prop =>
+    !['size', 'isDropdownVisible', 'state', 'openable'].includes(prop),
+})<StyledInputWrapperProps>`
+  display: flex;
+  height: ${({ size }) => INPUT_SIZE_HEIGHT[size]}px;
+  padding: ${({ theme }) => theme.space[1]};
+  padding-right: 0;
+  padding-left: ${({ theme }) => theme.space[2]};
+
+  background: ${({ theme }) => theme.colors.neutral.background};
+  border: 1px solid ${({ theme, state }) => theme.colors[state].border};
+  border-radius: ${({ theme }) => theme.radii.default};
+
+  &:hover,
+  :focus {
+    border-color: ${({ theme }) => theme.colors.primary.borderHover};
+    outline: none;
+  }
+  &[data-readonly='true'] {
+    background: ${({ theme }) => theme.colors.neutral.backgroundWeak};
+    border-color: ${({ theme }) => theme.colors.neutral.border};
+  }
+
+  &[data-disabled='true'] {
+    background: ${({ theme }) => theme.colors.neutral.backgroundDisabled};
+    border-color: ${({ theme }) => theme.colors.neutral.borderDisabled};
+  }
+
+  &:not([data-disabled='true']):not([data-readonly]):hover {
+    border-color: ${({ theme }) => theme.colors.primary.border};
+  }
+
+  &:active {
+    box-shadow: ${({ theme, openable }) =>
+      openable ? theme.shadows.focusPrimary : 'none'};
+  }
+
+  ${({ theme, isDropdownVisible }) =>
+    !isDropdownVisible
+      ? ''
+      : `
+          box-shadow: ${theme.shadows.focusPrimary};
+          border-color: ${theme.colors.primary.borderHover};
+          }
+        `}
+`
+const CustomTag = styled(Tag)`
+  height: fit-content;
+  width: fit-content;
+`
+
+const StyledPlaceholder = styled(Text)`
+  color: ${({ theme }) => theme.colors.neutral.textWeak};
+  text-size: ${({ theme }) => theme.typography.body.fontSize};
+  display: flex;
+  flex: 1;
+  align-self: center;
+`
+
+const DisplayValues = ({
+  multiselect,
+  refTag,
+  width,
+  nonOverflowedValues,
+  disabled,
+  readOnly,
+  selectedValues,
+  setSelectedValues,
+  onChange,
+  overflowed,
+  overflowAmount,
+}: DisplayValuesProps) =>
+  multiselect ? (
+    <StackTags
+      direction="row"
+      gap="1"
+      wrap="nowrap"
+      ref={refTag}
+      width={width}
+      alignItems="center"
+    >
+      {nonOverflowedValues.map((option, index) => (
+        <CustomTag
+          data-testid="selected-options-tags"
+          key={option ? option.value : index}
+          sentiment="neutral"
+          disabled={disabled}
+          onClose={
+            !readOnly
+              ? event => {
+                  event.stopPropagation()
+                  const newSelectedValues = selectedValues
+                    ?.filter(val => val !== option)
+                    .map(val => val?.value)
+                  setSelectedValues(
+                    selectedValues?.filter(val => val !== option),
+                  )
+                  onChange?.(newSelectedValues)
```

Comment:
I think it can be simplified to avoid doing the same filter twice, wdyt?

```tsx
                  const newSelected = selectedValues
                    ?.filter(val => val !== option)
                  setSelectedValues(newSelected)
                  onChange?.(newSelectedValues.map(val => val?.value))
```